go-controller/pkg/networkmanager/nad_controller.go

```diff
@@ -262,16 +333,24 @@ func (c *nadController) syncNAD(key string, nad *nettypes.NetworkAttachmentDefin
 
 	// remove the NAD reference from the old network and delete the network if
 	// it is no longer referenced
+	var releaseNetwork util.NetInfo
 	if oldNetwork != nil {
 		oldNetworkName := oldNetwork.GetNetworkName()
 		oldNetwork.DeleteNADs(key)
 		if len(oldNetwork.GetNADs()) == 0 {
 			c.networkController.DeleteNetwork(oldNetworkName)
+			releaseNetwork = oldNetwork
 		} else {
 			c.networkController.EnsureNetwork(oldNetwork)
+			// we have nothing else to do with oldNetwork so reset to nil
+			oldNetwork = nil
 		}
 	}
 
+	if err := c.handleNetworkID(releaseNetwork, ensureNetwork, nad); err != nil {
```

Comment:
So `zone == "" && node == ""` means its cluster manager because its not a zone nor a node. If that's the case, on the constructor we initialize the network ID allocator and from that point onward we use that to know if we are allocating network IDs or not. `handleNetworkID` gets the network ID and sets it on NetInfo. If there is no network ID and if we are allocating IDs, the we do so. If not, we will just wait until that happens.

Will add doc to the method to explain this further.